book/python/api.md

```diff
@@ -436,6 +480,52 @@ def extract_user(transaction):
     return transaction.user
 ```
 
+### Windows
+
+For an in-depth overview of windows, see [here](/book/core-concepts/windows.md). Windows are either count-based or range-based.
+
+#### Count-based
+
+`wallaroo.count_windows(count)`: defines count-based windows that trigger an output every `count` messages.
+
+`over(aggregation)`: specifies the aggregation used for these windows.
+
+##### Example
+
+This defines count windows with a count of 5 over a user-defined [aggregation](#aggregation) class `MySumAgg`:
+
+```python
+    (inputs
+        .to(wallaroo.count_windows(5)
+            .over(MySumAgg))
+        .to_sink(sink_config))
+```
+
+#### Range-based
+
+`wallaroo.range_windows(range)`: defines range-based windows with the specified `range`.
+
+`with_slide(slide)`: specifies a slide for these windows, resulting in sliding (overlapping) windows. For example, if the slide is 3 seconds, then a new window of length `range` (specified via `wallaroo.range_windows(range)`) will be started every 3 seconds.
+
+`with_delay(delay)`: specifies a delay on triggering a window. This is either (1) an estimation of the maximum lateness expected for messages or (2) the lateness threshold beyond which you no longer care about messages.
+
+`over(aggregation)`: specifies the aggregation used for these windows.
+
+##### Example
+
+This defines range-based sliding windows with a range of 6 seconds, a slide of 3 seconds, and a delay of 10 seconds:
+
+```python
+    (inputs
+        .to(wallaroo.range_windows(wallaroo.seconds(6))
+            .with_slide(wallaroo.seconds(3))
+            .with_delay(wallaroo.seconds(10))
+            .over(MySumAgg))
+        .to_sink(sink_config))
+```
+
+This means windows of length 6 seconds will be started every 3 seconds. Each of these windows will be triggered 10 seconds after we have past the end of its range in event time.
+
```

Comment:
are `with_delay` and `with_slide` optional? If so what are their defaults? This should be noted here.

Reply:
addressed